drawing_agent/app/drawing_cache.py

```
@@ -0,0 +1,96 @@
+        num_id = self._int_id(hash_id)
+        ocr_text = static_data.get("ocr_text", "")
+        if not ocr_text:
+            return
+        fragment = f"Распознанный текст чертежа: {ocr_text}"
+        vec = self.embed_model.encode(fragment, convert_to_numpy=True).astype('float32').reshape(1, -1)
+        self.vector_db.add(vec, [num_id], [{"type": "static", "text": fragment, "drawing_hash": hash_id}])
+
+    def add_interaction_to_index(self, path: str, page: int, question: str, answer: str):
+        hash_id = self._get_drawing_hash(path, page)
+        num_id = self._int_id(hash_id)
+        combined = f"Вопрос: {question}\nОтвет: {answer}"
```

Ruff (0.15.12)

[warning] String contains ambiguous `О` (CYRILLIC CAPITAL LETTER O). Did you mean `O` (LATIN CAPITAL LETTER O)?

(RUF001)